### PR TITLE
[WIP] Add "How does the mutant gem work" article.

### PR DESCRIPTION
**Pull request by troessner**
cc @mbj - I'll take your feedback from [here](https://github.com/troessner/articles/commit/a7be614a4aafb870f820f8a34aa2d7da17ee581e) into account later.

### CODE REVIEW

File: draft-how-does-the-mutant-gem-work.markdown [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Or this
+
+```Ruby
+if @enabled
+  "#{@phrase}#{" "}#{options.fetch(:name)}"
+end
+
+# Was
+# "#{@phrase} #{options[:name]}" if @enabled
+```
```
Comment (mbj) [inline draft-how-does-the-mutant-gem-work.markdown:51]: You may want to add the a trivial test gets 100% line coverage, where the branch condition is not covered, but evaluating to true.

**Comment by mbj** [comment]
@troessner Before starting with mutant, what about covering the ruby AST by `parser` it has very good documentation: https://github.com/whitequark/parser/blob/master/doc/AST_FORMAT.md

Rationale: before jumping into "how" mutant works, maybe explain the level of abstraction it works on?

you could describe on how blocks look like on the AST, method calls in their various flavors, op-assigns etc.

**Comment by troessner** [comment]
> > you could describe on how blocks look like on the AST, method calls in their various flavors, op-assigns etc.

Like the idea, but I think I'd rather do it in the context of Mutant - keep in mind, that I intend to write a blog series, not just a single one, so there should be enough room for explanation.

**Comment by troessner** [comment]
Closing this one in favor of #3

**Pull request closed without merging**